elasticdl/python/ps/parameters.py

```diff
@@ -110,3 +116,22 @@ def _init_non_embedding_params(self, tensors_pb):
     def _init_embedding_params(self, embeddings_pb):
         for pb in embeddings_pb:
             self.embedding_params[pb.name] = create_embedding_table(pb)
+
+    def has_embedding_params(self):
+        return len(self.embedding_params) > 0
+
+    def create_slot_params(self, slot_names, init_values):
+        embed_layer_names = list(self.embedding_params.keys())
+        for layer_name in embed_layer_names:
+            for slot_name in slot_names:
```

Comment:
One question, are the slots of all the embedding layers the same?

Reply:
Yes, it is decided by the optimizer type. Adam creates slots "m" and "v" for each variable, and  Adagrad creates slots "accumulator" for each variable. The user-defined arguments of initial values are also in optimizer.